Read PROJECT and REGION from Colab secrets or environment variables

tylere opened the pull request:
Changes
- Incorporate Nick's updates from [this notebook](https://colab.research.google.com/drive/1NCaPOoxqmAEWb8c8V0kEHVunbW1yHVkL?resourcekey=0-HJ3ou94AbjdKkkvaPW1Jtw&usp=sharing#scrollTo=1p3MKGcXDAEr)
- Remove hardcoding of project name
- Read PROJECT and REGION from Colab secrets or local environment variables

tylere committed:
Replace with latest notebook version

tylere committed:
Clear all outputs

tylere committed:
Stop hardcoding the project name

tylere committed:
Read PROJECT and REGION from Colab secrets or environment vars

tylere committed:
Add comments and fix variable naming

tylere committed:
Merge branch 'main' into whisp-cloud-function

n-clinton reviewed (changes requested):
If I click "view file"I get an Invalid Notebook error.  Should I be worried about that?

brycecr reviewed (changes requested):
as Nick mentioned, please tweak so the notebook is considered valid by github. https://github.com/tylere/forest-data-partnership/blob/whisp-cloud-function/notebooks/WHISP_and_chains_the_easy_way.ipynb is currently not valid

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
```diff
@@ -1,104 +1,179 @@
       "source": [
-        "!gcloud auth login --project {PROJECT} --billing-project {PROJECT} --update-adc"
-      ],
+        "def is_running_in_colab():\n",
```
imports in python are bound to the current scope, so this does not work as you intend in colab, unfortunately.

brycecr replied:
possibly could just have different options depending on whether it's run in colab or not and describe the difference so a reader can do it correctly if there isn't an easy one-size-fits-all approach

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
```diff
@@ -110,7 +185,7 @@
         "    scopes=['https://www.googleapis.com/auth/earthengine']\n",
         ")\n",
-        "ee.Initialize(credentials, project='forest-data-partnership', opt_url='https://earthengine-highvolume.googleapis.com')\n",
+        "ee.Initialize(credentials, project=PROJECT, opt_url='https://earthengine-highvolume.googleapis.com')\n",
```
this will just write PROJECT into the file, not the value it holds. That's fine but will lead to errors on turn-up for the cloud function. Consider adding a comment instructing the user to replace PROJECT? Otherwise you can write a custom magic function that supports parameters and get a the desired result, e.g. with https://github.com/nopdotcom/jupyter-jinja-yaml-magic or [this hack](https://github.com/ipython/ipython/issues/6701#issuecomment-382640776). Note that you probably need the surrounding quotes too.

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
```diff
@@ -188,7 +272,7 @@
         "      scopes=['https://www.googleapis.com/auth/earthengine']\n",
         "  )\n",
-        "  ee.Initialize(credentials, project='forest-data-partnership')\n",
+        "  ee.Initialize(credentials, project=PROJECT)\n",
```
same PROJECT issue here as above

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
this is now duplicated above, before the SQL section starts. Probably keep the one above and drop this?

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
actually I think this whole section is duplicated

brycecr commented on notebooks/WHISP_and_chains_the_easy_way.ipynb:
````diff
@@ -298,166 +423,204 @@
-        "test_request = json.dumps({'calls': test_calls}, separators=(',', ':')).join(\"''\")"
-      ],
+        "```\n",
+        "CREATE OR REPLACE FUNCTION `forest-data-partnership.WHISP_DEMO.whisp`(geom STRING) RETURNS STRING\n",
````
suggest replacing all instances of 'forest-data-partnership' with PROJECT and either instructing the user to input their own project, or generate the command from template

brycecr reviewed (changes requested):
This is really cool! I was able to successfully go through the colab with minor tweaks that I have indicated here.

tylere committed:
Saved after using "Clear all outputs" to address GitHub's Invalid Notebook message

Invalid Notebook
Additional properties are not allowed ('execution_count', 'outputs' were unexpected)
Using nbformat v5.10.4 and nbconvert v7.16.6

tylere commented:
> as Nick mentioned, please tweak so the notebook is considered valid by github. https://github.com/tylere/forest-data-partnership/blob/whisp-cloud-function/notebooks/WHISP_and_chains_the_easy_way.ipynb is currently not valid

This was resolved by Colab -> Edit -> Clear all outputs

tylere commented:
@brycecr this has been rebased with main, simplifying the file changes.

brycecr reviewed (approved):
Approving, I can make minor updates post merge

n-clinton reviewed (approved):
Approved
